pkg/network/usm/utils/file_registry.go

```diff
@@ -84,13 +84,19 @@ func NewFileRegistry(programName string) *FileRegistry {
 		log.Warnf("running without block cache list, creation error: %s", err)
 		blocklistByID = nil
 	}
-	return &FileRegistry{
+	r := &FileRegistry{
 		procRoot:      kernel.ProcFSRoot(),
 		byID:          make(map[PathIdentifier]*registration),
 		byPID:         make(map[uint32]pathIdentifierSet),
 		blocklistByID: blocklistByID,
 		telemetry:     newRegistryTelemetry(programName),
 	}
+
+	// Add self to the debugger so we can inspect internal state of this
+	// FileRegistry using our debugging endpoint
+	debugger.Add(r)
```

Comment:
when do we delete from the debugger?
the concern is having an ever increasing "in memory storage" for FileRegistry

Reply:
wondering if we should have the `debugger` live all the time
a different approach would be - when triggering the endpoint, then iterate over all FileRegistries, and in the end, delete (the GC) the resources
that way we don't introduce a potential "leak"

Reply:
The debugger shouldn't do any long-lived memory allocations, so the memory footprint should be virtually zero.
The only thing it keeps throughout the whole lifecycle of system-probe is one pointer to each file registry.